src/application-page-status/ApplicationPageStatusProvider.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import Scroll from 'terra-scroll';
+import StatusView from 'terra-status-view';
+import ApplicationPageStatusContext from './ApplicationPageStatusContext';
+import styles from './ApplicationPageStatusProvider.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The components to be rendered within the context of the ApplicationPageStatusProvider.
+   * Components rendered here are able to interact with ApplicationPageStatusProvider through
+   * the ApplicationPageStatusContext.
+   */
+  children: PropTypes.node,
+  /**
+   * A function to be called with the current ref of the scrollable element rendered within the
+   * ApplicationPageStatusProvider.
+   */
+  scrollRefCallback: PropTypes.func,
+};
+
+const ApplicationPageStatusProvider = ({ children, scrollRefCallback, ...customProps }) => {
+  const [registeredPageStatus, setRegisteredPageStatus] = React.useState({});
+
+  const contextValue = useMemo(() => ({
+    show: (key, data) => {
+      setRegisteredPageStatus(state => (
+        {
+          ...state,
+          [`${key}`]: data,
+        }
+      ));
+    },
+    hide: (key) => {
+      setRegisteredPageStatus((state) => {
+        const newRegisteredPageStatus = { ...state };
+        delete newRegisteredPageStatus[key];
+
+        return newRegisteredPageStatus;
+      });
+    },
+  }), []);
+
+  const registeredPageStatusKeys = Object.keys(registeredPageStatus);
+  const registeredButtonAttrs = registeredPageStatusKeys.map(key => (registeredPageStatus[key] && registeredPageStatus[key].buttonAttrs));
+  const registeredMessage = registeredPageStatusKeys.map(key => (registeredPageStatus[key] && registeredPageStatus[key].message));
+  const registeredTitle = registeredPageStatusKeys.map(key => (registeredPageStatus[key] && registeredPageStatus[key].title));
+  const registeredVariant = registeredPageStatusKeys.map(key => (registeredPageStatus[key] && registeredPageStatus[key].variant));
+
```

Comment:
These repeated maps seem inefficient. Would it make more sense to extract the data you'll use for presentation into a separate state key to limit the logic necessary to re-render?

Reply:
i agree, so i have made the necessary changes here : https://github.com/cerner/terra-application/pull/47/commits/108d8ce2f7edd49a0a770fdb02a18227e2239c77